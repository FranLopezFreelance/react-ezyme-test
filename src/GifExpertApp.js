import React, { Fragment, useState } from 'react'
import PropTypes from 'prop-types';
import AddCategory from './components/AddCategory';
import { GifsGrid } from './components/GifsGrid';

const GifExpertApp = ({name}) => {

  const [categories, setCategory] = useState(['One Punch']);

  return (
    <Fragment>
      <h2 className="animate__animated animate__fadeIn">{name}</h2>
      <AddCategory setCategory={setCategory} />
      <hr />
      <ul>
        { categories.map(category => 
            <GifsGrid 
              key={category} 
              category={category} />
          ) }
      </ul>
    </Fragment>
  )
}

GifExpertApp.propTypes = {
  name: PropTypes.string.isRequired
}

export default GifExpertApp;
