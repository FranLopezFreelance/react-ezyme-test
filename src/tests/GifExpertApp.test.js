import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el <GifExpertApp />', () => {
  const name = 'Gif Expert App';

  test('Debe renderizarse el componente', () => {
    const wrapper = shallow(<GifExpertApp name={name} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar una lista de categorÃ­as', () => {
    const categories = ['One Punch', 'Los SImpsons'];
    const wrapper = shallow(<GifExpertApp name={name} defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifsGrid').length).toBe(categories.length);
  });
});