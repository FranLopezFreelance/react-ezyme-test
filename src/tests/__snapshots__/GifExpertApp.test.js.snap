// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul>
    <GifsGrid
      category="One Punch"
      key="One Punch"
    />
    <GifsGrid
      category="Los SImpsons"
      key="Los SImpsons"
    />
  </ul>
</Fragment>
`;

exports[`Pruebas en el <GifExpertApp /> Debe renderizarse el componente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
